code/modules/clothing/accessories/rings.dm

```diff
@@ -0,0 +1,201 @@
+//Generic Ring
+
+/obj/item/clothing/accessory/ring
+	name = "generic ring"
+	desc = "Torus shaped finger decoration."
+	icon_state = "material"
+	drop_sound = 'sound/items/drop/ring.ogg'
+	slot = ACCESSORY_SLOT_RING
+	slot_flags = SLOT_GLOVES
+	item_icons = list(
+		slot_l_hand_str = 'icons/mob/items/lefthand_gloves.dmi',
+		slot_r_hand_str = 'icons/mob/items/righthand_gloves.dmi',
+		)
+	gender = PLURAL
+	w_class = ITEMSIZE_SMALL
+	icon = 'icons/inventory/hands/item.dmi'
+	drop_sound = 'sound/items/drop/ring.ogg'
+	pickup_sound = 'sound/items/pickup/ring.ogg'
+
+/////////////////////////////////////////
+//Standard Rings
+/obj/item/clothing/accessory/ring/engagement
+	name = "engagement ring"
+	desc = "An engagement ring. It certainly looks expensive."
+	icon_state = "diamond"
+
+/obj/item/clothing/accessory/ring/engagement/attack_self(mob/user)
+	user.visible_message(span_warning("\The [user] gets down on one knee, presenting \the [src]."),span_warning("You get down on one knee, presenting \the [src]."))
+
+/obj/item/clothing/accessory/ring/cti
+	name = "CTI ring"
+	desc = "A ring commemorating graduation from CTI."
+	icon_state = "cti-grad"
+
+/obj/item/clothing/accessory/ring/mariner
+	name = "Mariner University ring"
+	desc = "A ring commemorating graduation from Mariner University."
+	icon_state = "mariner-grad"
+
+
+/////////////////////////////////////////
+//Reagent Rings
+
+/obj/item/clothing/accessory/ring/reagent
+	flags = OPENCONTAINER
+	origin_tech = list(TECH_MATERIAL = 2, TECH_ILLEGAL = 4)
+
+/obj/item/clothing/accessory/ring/reagent/New()
+	..()
+	create_reagents(15)
+
+/obj/item/clothing/accessory/ring/reagent/equipped(var/mob/living/carbon/human/H)
+	..()
+	if(istype(H) && H.gloves==src)
+
+		if(reagents.total_volume)
+			to_chat(H, span_danger("You feel a prick as you slip on \the [src]."))
+			if(H.reagents)
+				var/contained = reagents.get_reagents()
+				var/trans = reagents.trans_to_mob(H, 15, CHEM_BLOOD)
+				add_attack_logs(usr, H, "Injected with [name] containing [contained] transferred [trans] units")
+	return
+
+//Sleepy Ring
+/obj/item/clothing/accessory/ring/reagent/sleepy
+	name = "silver ring"
+	desc = "A ring made from what appears to be silver."
+	icon_state = "material"
+	origin_tech = list(TECH_MATERIAL = 2, TECH_ILLEGAL = 5)
+
+/obj/item/clothing/accessory/ring/reagent/sleepy/New()
+	..()
```

Comment:
This likely can use Initialize without issues

Initialize()
. = ..()

then the add reagents. We want to get away from NEW usages